src/analyses/apron/octApronPriv.apron.ml

```diff
@@ -416,8 +416,27 @@ struct
         | _ ->
           st
       end
+    | `Join ->
+      if (ask.f Q.MustBeSingleThreaded) then
+        st
+      else
+        let oct = st.oct in
+        let g_vars = List.filter (fun var ->
+            match V.find_metadata var with
+            | Some (Global _) -> true
+            | _ -> false
+          ) (AD.vars oct)
+        in
+        let oct_side = AD.keep_vars oct g_vars in
+        sideg (mutex_inits ()) oct_side;
```

Comment:
Inspired by `enter_multithreaded` I suppose it makes sense to also side effect to `mutex_inits` here, but curiously the analogous case in basePriv doesn't do this:
https://github.com/goblint/analyzer/blob/9fd923b76a443e2f9e08d1f68f124d6195ed0264/src/analyses/basePriv.ml#L370-L383

Should basePriv mutex-meet also be doing that then?

Reply:
This is a broader issue tracked by #327.